docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerCloud.java

```diff
@@ -269,10 +265,11 @@ public Node call() throws Exception {
                                     //
                                     // deferring the completion of provisioning until the launch
```

Comment:
That's the general overview of it.

The gotcha - if there is one - is that in some ways the easiest way to test if the node is ready is to try to connect with a launcher... which needs the node to be added to Jenkins so you can get the other half of the Node/Computer duality created.

The correct way is to use a non-launcher based way to determine that the node is ready to connect with